src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputerLauncher.java

```diff
@@ -59,6 +64,34 @@ public static void log(
     }
   }
 
+  protected void log(
+      Level level, ComputeEngineComputer computer, TaskListener listener, String message) {
+    try {
+      ComputeEngineCloud cloud = computer.getCloud();
+      cloud.log(LOGGER, level, listener, message);
+    } catch (CloudNotFoundException cnfe) {
+      log(LOGGER, Level.SEVERE, listener, "FATAL: Could not get cloud");
+    }
+  }
+
+  protected void logException(
+      ComputeEngineComputer computer, TaskListener listener, String message, Throwable exception) {
+    try {
+      ComputeEngineCloud cloud = computer.getCloud();
```

Comment:
There seems to be duplicated logic here between log and logException. Suggest having logException and others call log().

Reply:
Done.